crates/sui-graphql-rpc/src/types/name_service_name.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::*;
+use move_core_types::language_storage::StructTag;
+use serde::{Deserialize, Serialize};
+use sui_json_rpc::name_service::Domain;
+
+use super::move_object::MoveObject;
+
+#[derive(Clone, Serialize, Deserialize)]
+pub(crate) struct NativeSuinsRegistration {
+    pub id: sui_types::id::UID,
+    pub domain: Domain,
+    pub domain_name: String,
+    pub expiration_timestamp_ms: u64,
+    pub image_url: String,
+}
+
+#[derive(Clone)]
+pub(crate) struct SuinsRegistration {
+    /// Representation of this SuinsRegistration as a generic Move object.
+    pub super_: MoveObject,
+
+    /// The deserialized representation of the Move object's contents.
+    pub native: NativeSuinsRegistration,
+}
+
+pub(crate) enum SuinsRegistrationDowncastError {
+    NotASuinsRegistration,
+    Bcs(bcs::Error),
+}
+
+#[Object]
+impl SuinsRegistration {
+    /// Domain name of the SuinsRegistration object
+    async fn domain(&self) -> &str {
+        &self.native.domain_name
+    }
+
+    /// Convert the SuinsRegistration object into a Move object
+    async fn as_move_object(&self) -> &MoveObject {
+        &self.super_
+    }
+}
+
+impl TryFrom<(&MoveObject, &StructTag)> for SuinsRegistration {
```

Comment:
Don't implement conversion traits against tuples -- that's just a regular function that you've made difficult to call!

Reply:
But at the same time, I don't fully understand why you need to pass the `StructTag` here?  Surely that would be a constant that represents the `SuinsRegistration` type?

Reply:
@manolisliolios iiuc the mainnet package address for suins is fixed, but testnet is not and it changes after every testnet wipe?

Reply:
Ah got it -- that is annoying, but okay, then this shouldn't be a `TryFrom` impl, it should just be a function:

```
impl SuinsRegistration {
    fn try_from(object: &MoveObject, tag: &StructTag) -> Result<Self, Error> { /* ... */ }
}
```